src/DetailsView/components/requirement-view-component-configuration.tsx

```diff
@@ -7,9 +7,21 @@ import {
     getNextRequirementConfigurationForAssessment,
     getNextRequirementConfigurationForQuickAssess,
 } from 'DetailsView/components/requirement-view-next-requirement-configuration';
+import {
+    ShouldShowRequirementContextBox,
+    shouldShowRequirementContextBoxForAssessment,
+    shouldShowRequirementContextBoxForQuickAssess,
+} from 'DetailsView/components/requirement-view-should-show-context-box';
+import {
+    ShouldShowInfoButton,
+    shouldShowInfoButtonForAssessment,
+    shouldShowInfoButtonForQuickAssess,
+} from 'DetailsView/components/requirement-view-should-show-info-button';
 
 export type RequirementViewComponentConfiguration = {
     getNextRequirementButtonConfiguration: GetNextRequirementButtonConfiguration;
+    shouldShowInfoButton: ShouldShowInfoButton;
```

Comment:
I think we could simplify the components that are using these flags by instead returning the components that we want instead (as the switcher configuration is already operating as a logical switch anyways). As in to say this becomes something like:

```infoButton: ReactFCWithDisplayName<some-shared-props-type>```
or alternatively
```getShowInfoButton: (props: some-shared-props-type) => JSX.Element```

would then mean components go from:

``` 
{props.shouldRenderInfoButton && <info button component>
```
to
```
const InfoButton = config.infoButton;*
<InfoButton {...props}>
``` 
reducing the need for a logical check.

\* (variable assignment is only necessary because I believe TS has an issue with using the object inline for component declaration)